api/src/test/java/org/openmrs/module/appointments/service/impl/TeleconsultationAppointmentServiceTest.java

```diff
@@ -23,21 +34,60 @@
     @Mock
     private AdministrationService administrationService;
 
+    @Mock
+    private PatientService patientService;
+
+    @Mock
+    private PatientAppointmentNotifierService patientAppointmentNotifierService;
+
+    @InjectMocks
+    private TeleconsultationAppointmentService teleconsultationAppointmentService;
+
+    private Patient patient;
+
     @Before
     public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+        patient = new Patient();
+        patient.setUuid("patientUuid");
+        PersonName name = new PersonName();
+        name.setGivenName("test patient");
+        Set<PersonName> personNames = new HashSet<>();
+        personNames.add(name);
+        patient.setNames(personNames);
+        PatientIdentifier identifier = new PatientIdentifier();
+        identifier.setIdentifier("GAN230901");
+        PatientIdentifierType patientIdentifierType = new PatientIdentifierType();
+        patientIdentifierType.setName("Patient Identifier");
+        identifier.setIdentifierType(patientIdentifierType);
+        patient.setIdentifiers(new HashSet<>(Arrays.asList(identifier)));
+        when(patientService.getPatientByUuid("patientUuid")).thenReturn(patient);
         PowerMockito.mockStatic(Context.class);
         when(Context.getAdministrationService()).thenReturn(administrationService);
         when(administrationService.getGlobalProperty("bahmni.appointment.teleConsultation.serverUrlPattern")).thenReturn("https://test.server/{0}");
+        when(administrationService.getGlobalProperty("bahmni.adhoc.teleConsultation.id")).thenReturn("Patient Identifier");
     }
 
     @Test
     public void shouldGenerateTCLinkForAppointment() {
-        TeleconsultationAppointmentService teleconsultationAppointmentService = new TeleconsultationAppointmentService();
         Appointment appointment = new Appointment();
         UUID uuid = UUID.randomUUID();
         appointment.setUuid(uuid.toString());
         String link = teleconsultationAppointmentService.generateTeleconsultationLink(appointment);
         assertEquals("https://test.server/"+appointment.getUuid(), link);
 
     }
+
+    @Test
+    public void shouldGetAdhocTCLink() {
```

Comment:
I think this test name should be: shouldGetAdhocTCLinkBasedOnPatientID( ). What do you feel?

Reply:
Yeah forgot to change the name after changing implementation to use patient id, changed now